Position mode stick (#1158)
PX4/PX4-user_guide: en/flight_modes/position_mc.md

[<img src="../../assets/site/difficulty_easy.png" title="Easy to fly" width="30px" />](../getting_started/flight_modes.md#key_difficulty)&nbsp;[<img src="../../assets/site/remote_control.svg" title="Manual/Remote control required" width="30px" />](../getting_started/flight_modes.md#key_manual)&nbsp;[<img src="../../assets/site/position_fixed.svg" title="Position fix required (e.g. GPS)" width="30px" />](../getting_started/flight_modes.md#key_position_fixed)

*Position* is an easy-to-fly RC mode in which roll and pitch sticks control acceleration over ground in the vehicle's left-right and forward-back directions (similar to a car's accelerator pedal), and throttle controls speed of ascent-descent.

> When the sticks are released/centered the vehicle will actively brake, level, and be locked to a position in 3D space — compensating for wind and other forces.

Reviewer: ```suggestion
When the sticks are released/centered the vehicle will actively brake, level, and be locked to a position in 3D space — compensating for wind and other forces.
With full stick deflection the vehicle accelerates initially with [MPC_ACC_HOR_MAX](../advanced_config/parameter_reference.md#MPC_ACC_HOR_MAX) ramping down until it reaches the final velocity [MPC_VEL_MANUAL](../advanced_config/parameter_reference.md#MPC_VEL_MANUAL).
```

:::tip
Position mode is the safest manual mode for new fliers. 
Unlike [Altitude](../flight_modes/altitude_mc.md) and [Manual/Stabilized](../flight_modes/manual_stabilized_mc.md) modes the vehicle will stop when the sticks are centered rather than continuing until slowed by wind resistance.
:::

The diagram below shows the mode behaviour visually (for a mode 2 transmitter).

![MC Position Mode](../../assets/flight_modes/position_MC.png)

> :::warning

Reviewer: ```suggestion
```

> Care must be taken when landing in this mode.

Author: @MaEtUgR Not sure about this note - it seems pretty scary for "the safest mode on PX4". 

My assumption is that this is here as a warning for **first-time flying** a vehicle that might not have landing detector set up right? So that part isn't such a risk if you have landing detector working?

Further, the GPS drift thing - when would that happen, and how risky is it? 

If this is all low risk then I might add a section on "Landing" for this stuff.

> Care must be taken when landing in this mode.
> When first landing in this mode, be ready to switch to [Manual/Stabilized](../flight_modes/manual_stabilized_mc.md) in order to be able to disarm.
> If landing is correctly detected, motors will spin down after touch down and then disarm shortly after.
> If the motors keep spinning at higher RPM or start spinning up, first switch to [Manual/Stabilized (MC)](../flight_modes/manual_stabilized_mc.md), and then disarm.
> Be aware that the vehicle may tip over on the ground due to GPS drift.

Reviewer: Taking an attempt at explaining landing. I know there's a wrapping warning but GitHub wouldn't let me edit it at the same time.
```suggestion
### Landing

Landing in this mode is easy:
1. Position the drone horizontally above the landing spot using the roll and pitch stick.
1. Let go of the roll and pitch stick and give it enough time to come to a complete stop.
1. Pull the throttle stick down gently until the vehicle touches the ground.
1. Pull the throttle stick all the way down to facilitate and accelerate land detection.
1. The vehicle will lower propeller thrust, detect the ground and automatically disarm by default (see [COM_DISARM_LAND](../advanced_config/parameter_reference.md#COM_DISARM_LAND))

:::warning
In case the vehicle fails to stop horizontally: Keep it under control in [Altitude](../flight_modes/altitude_mc.md) mode and land like instructed above but manually making sure the vehicle stays above the landing spot using roll and pitch stick. After landing check GPS and magnetometer orientation, calibration.

In case the vehicle fails to detect the ground and landing: After the vehicle is on the ground switch to [Manual/Stabilized](../flight_modes/manual_stabilized_mc.md) mode keeping the throttle stick low and manually disarm using a gesture or other command. Alternatively you can also use the kill switch when the vehicle is already on the ground.
:::
```

> :::

Reviewer: ```suggestion
```

## Technical Summary

RC/manual mode where RPT sticks control *speed* in corresponding directions. Centered sticks level vehicle and hold it to fixed position and altitude against wind.

> * Centered RPT sticks (in RC deadzone) hold x, y, z position steady against any wind, and levels attitude.

Reviewer: ```suggestion
* Centered roll, pitch, throttle sticks (within RC deadzone [MPC_HOLD_DZ](../advanced_config/parameter_reference.md#MPC_HOLD_DZ)) hold x, y, z position steady against any disturbance like wind.
```

* Outside center:

>   * Roll/Pitch sticks control horizontal acceleration (over ground) in left-right and forward-back directions (respectively) relative to the "front" of the vehicle.

Reviewer: ```suggestion
  * Roll/Pitch sticks control horizontal acceleration over ground in forward-back and left-right directions relative to the front direction of the vehicle.
```

1. The vehicle will lower propeller thrust, detect the ground and automatically disarm by default (see [COM_DISARM_LAND](../advanced_config/parameter_reference.md#COM_DISARM_LAND))

:::warning
While very rare on a well calibrated vehicle, sometimes there may be problems with landing.
- If the vehicle does not stop moving horizontally:
  - You can still land under control in [Altitude mode](../flight_modes/altitude_mc.md).
    The approach is the same as above, except that you manually ensure that the vehicle stays above the landing spot using the roll and pitch stick.
  - After landing check GPS and magnetometer orientation, calibration.

> - If the vehicle does not detect the ground/landing and disarm:

Reviewer: Probably it would be helpfull to link to https://docs.px4.io/master/en/advanced_config/prearm_arm_disarm.html#auto-disarming on the words "automatically disarm". Gives you the right pointers to continue reading how it works.

Author: Thanks. Agree. Fixed in another PR.

  - After the vehicle is on the ground switch to [Manual/Stabilized mode](../flight_modes/manual_stabilized_mc.md) keeping the throttle stick low, and manually disarm using a gesture or other command.
    Alternatively you can also use the kill switch when the vehicle is already on the ground.
:::

## Technical Summary

RC mode where roll, pitch, throttle (RPT) sticks control movement in corresponding axes/directions.

> Centered sticks level vehicle and hold it to fixed altitude and position against wind.

Reviewer: It's probably just technical wise guy nit-pick but nothing guarantees you the vehicle will stay level. It will correct velocity and position and in the desired steady-state case where it arrived very close to the goal position and there's no external disturbance like strong wind it's nearly level but still does some corrections. I think you can judge better how users will intuitively understnad it.

Author: @MaEtUgR Hmmm. The point we're trying to make is that if the sticks are centered the vehicle _attempts_ to hold position and attitude. I don't think we're saying that it will always be perfectly level/not do corrections. 

That said, if you think it more accurate we could change:

> Centered sticks level vehicle and hold it to fixed altitude and position against wind.

To

> Centered sticks level the vehicle and actively brake, then and attempt to hold it it's altitude, position and attitude (compensating for wind or other external forces).

Reviewer: Thanks for the attempt. That's probably unnecessarily complicated and still has "level the vehicle" so let's just leave it as is. Understandable > wise guy correctness. 😇

<span id="MPC_LAND_ALT2"></span>[MPC_LAND_ALT2](../advanced_config/parameter_reference.md#MPC_LAND_ALT2) | Altitude for second phase of slow landing. In this phase maximum horizontal velocity is limited to [MPC_LAND_VEL_XY](#MPC_LAND_VEL_XY). Default 5m.
<span id="RCX_DZ"></span>`RCX_DZ` | RC dead zone for channel X. The value of X for throttle will depend on the value of [RC_MAP_THROTTLE](../advanced_config/parameter_reference.md#RC_MAP_THROTTLE). For example, if the throttle is channel 4 then  [RC4_DZ](../advanced_config/parameter_reference.md#RC4_DZ) specifies the deadzone.
<span id="MPC_xxx"></span>`MPC_XXXX` | Most of the MPC_xxx parameters affect flight behaviour in this mode (at least to some extent). For example, [MPC_THR_HOVER](../advanced_config/parameter_reference.md#MPC_THR_HOVER) defines the thrust at which a vehicle will hover.

> <span id="MPC_POS_MODE"></span>[MPC_POS_MODE](../advanced_config/parameter_reference.md#MPC_POS_MODE) | Stick control mode for movement over ground. From PX4 v1.12 the default (4) is that stick position controls acceleration (in a similar way to a car accelerator pedal). Other options allow stick angle to directly control speed over ground, with and without smoothing/acceleration limits.

Reviewer: ```suggestion
<span id="MPC_POS_MODE"></span>[MPC_POS_MODE](../advanced_config/parameter_reference.md#MPC_POS_MODE) | Stick input to movement translation strategy. From PX4 v1.12 the default (4) is that stick position controls acceleration (in a similar way to a car accelerator pedal). Other options allow stick deflection to directly control speed over ground, with and without smoothing and acceleration limits.
```
